src/indexer/utils/substrate/index.ts

```diff
@@ -124,10 +114,9 @@ export async function saveDeposit(
   } = substrateDepositData
 
   const decodedAmount = getDecodedAmount(depositData)
-
-  const foundTransfer = await transferRepository.findByNonceFromDomainId(Number(depositNonce), `${originDomainId}`)
-
-  if (foundTransfer) {
+  let transfer: Transfer | null
+  transfer = await transferRepository.findByNonceFromDomainId(Number(depositNonce), `${originDomainId}`)
```

Comment:
```suggestion
  let transfer = await transferRepository.findByNonceFromDomainId(Number(depositNonce), `${originDomainId}`)
```

Reply:
This will create a new transfer I want the same one for the transfer.id